src/components/policy/policy_external/src/cache_manager.cc

```diff
@@ -2629,6 +2629,12 @@ bool CacheManager::Init(const std::string& file_name,
       LOG4CXX_DEBUG(logger_,
                     "Check if snapshot valid: " << std::boolalpha << result);
 
+      result &= UnwrapAppPolicies(pt_->policy_table.app_policies_section.apps);
```

Comment:
this doesn't make sense, I don't think you can use the same result variable here as it is used for later logic.

I would just put this function call inside the if condition right below.

Reply:
Unwrapping policies is preparation step for policies data, In case if it is unsuccessful for some reason we should understand it as wrong policies data, and follow corresponding strategy. I agree that return on `UnwrapAppPolicies = false` is redundant and probably even malicious. I would keep appropriate error message in the `if` and then follow to `else` statement for result checking.

Reply:
That sounds fine to me, and so does the return void idea because there is also already a log in the fail case of `UnwrapAppPolicies`. Personally I would leave the function as bool and also log here.

I just don't think we should change the result variable as it would create a case where we lose functionality.

Reply:
Thank you for review, I've implemented the proposed changes in commits 4b62a2c03b898dd096d50b38ea1dc15227b973c9 and 7412f0c07518e6e21d7bea98c813cad7c28d3330.

Reply:
Additionally I've made some changes for external flow in commit 6f62d19ed04e987f41e0c24e66c1ec388955a9f9

Reply:
@ychernysheva thank you for your updates.

@LuxoftAKutsan I apologize, I did not understand your message correctly when I last replied. I see now that you did propose to do `result &=` with the return value, but I am still not convinced this is a good idea.

A policy table could still exist with an app policy: `"default": null,`, but I don't think it is okay to add a requirement that this key exists at this time. I am weary of changing behavior like this just because, no matter how unlikely, it is possible someone is using a policy table without the default key. I doubt anyone is doing this, but I don't see any advantage to taking the risk of breaking their flow.

If there is anything that would be broken by having the "default" key missing in app policies, please let me know and I will reconsider but otherwise this requirement should not be added.

Reply:
@iCollin  there is no problem if default is absent in policy table. 
But if default is absent and there are some application with 
`app: "default"`. that is the issue, and UnwrapAppPolicies should return false and SDL. Such policy table is invalid. but `is_valid` won't signal about it.

Reply:
@iCollin I've changed code in external flow in commit b877b37e3dd1b1fe337530e769d64aba04273069, and now it's similar to fix in proprietary flow. I hope that now it's correct from your perspective? If it's not, please request additional changes, and I'll be happy to implement them.

Reply:
@iCollin And I also want to propose squashing this commits before merging, because history of this review isn't so important. I think that it may be good idea, isn't it?